package/kedro_viz/models/flowchart.py

```diff
@@ -804,6 +751,25 @@ def set_preview(cls, _):
             )
             return None
 
+    @validator("preview_type", always=True)
+    def set_preview_type(cls, _):
+        if not (
+            not cls.data_node.disable_preview() and hasattr(cls.dataset, "preview")
+        ):
+            return None
```

Comment:
Again the double `not` makes this hard to read, I'd suggest changing it as above.